ballerina/channel_csv_io.bal

```diff
@@ -135,3 +127,87 @@ isolated function getWritableCSVChannel(WritableChannel writableChannel) returns
     }
     return writableCSVChannel;
 }
+
+isolated function readHeadersFromCsvFile(string path) returns string[]|Error {
+    stream<string[], Error?>|Error csvContent = fileReadCsvAsStream(path);
+    if csvContent is Error {
+        if csvContent is FileNotFoundError {
+            return [];
+        } else {
+            return error GenericError("Error while reading the headers from the CSV file. " + csvContent.message());
+        }
+    } else {
+        do {
+            var csvLine = check csvContent.next();
+            return csvLine == () ? [] : csvLine["value"];
+        } on fail Error err {
+            check csvContent.close();
+            return error GenericError("Error while reading the headers from the CSV file. " + err.message());
+        }
+    }
+}
+
+isolated function validateCsvHeaders(string[] headersFromCSV, string[] headers) returns string[]|Error {
+    if headersFromCSV == [""] || headersFromCSV.length() == 0 { //this is a valid scenario and we need to override the content.
+        return [];
+    } else if headers.length() != headersFromCSV.length() {
+        return error GenericError(string `The CSV file content header count(${headersFromCSV.length()}) doesn't match with ballerina record field count(${headers.length().toString()}). `);
+    }
+    foreach string header in headers {
+        int|Error? key = headersFromCSV.indexOf(header.trim());
+        if key is Error || key is () {
+            return error GenericError(string `The CSV file does not contain the header - ${header.trim()}. `);
+        }
+    }
+    return headersFromCSV;
+}
+
+isolated function appendRecordsToCsvFile(string path, map<anydata>[] contentToWrite, string[] headers) returns Error? {
+
+    WritableCSVChannel csvChannel = check getWritableCSVChannel(check openWritableCsvFile(path, option = APPEND));
+    foreach map<anydata> row in contentToWrite {
+        string[] sValues = [];
+        foreach string header in headers {
+            sValues.push(row.get(header).toString());
+        }
+        Error? writeResult = csvChannel.write(sValues);
+        if writeResult is Error {
+            check csvChannel.close();
+            return writeResult;
+        }
+    }
+    check csvChannel.close();
+}
+
+isolated function writeRecordsToCsvFile(string path, map<anydata>[] contentToWrite, string[] headers) returns Error? {
+    WritableCSVChannel csvChannel = check getWritableCSVChannel(check openWritableCsvFile(path, option = OVERWRITE));
```

Comment:
Lets pass the user provided option here. This function is being called for both cases, APPEND as well as OVERWRITE.

Reply:
updated in [35a6866](https://github.com/ballerina-platform/module-ballerina-io/pull/827/commits/35a6866f0651b523e84b3c608564fb09c0dafaa3)